### PR TITLE
Misc build script fixes for vendor platforms

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: build_tools/sdk.py [hunk 1]
```diff
@@ -510,7 +510,10 @@ def check_defold_sdk(sdkfolder, host_platform, platform, verbose=False):
     folders = []
     log_verbose(verbose, f"check_defold_sdk: {platform} {sdkfolder}")
 
-    if platform in ('x86_64-macos', 'arm64-macos', 'arm64-ios', 'x86_64-ios'):
+    if sdk_vendor is not None and sdk_vendor.supports_platform(platform):
+        folders = sdk_vendor.get_defold_sdk_folders(platform)
```
Comment (JCash) [inline build_tools/sdk.py:514]: Support for the new sdk_vendor.py file

File: engine/engine/src/wscript [hunk 2]
```diff
@@ -135,7 +135,7 @@ def build(bld):
         additional_libs.append('APP')
 
     platform_lib = ['PLATFORM']
-    if not platform_supports_feature(bld.env.PLATFORM, "opengl", None):
+    if not (platform_supports_feature(bld.env.PLATFORM, "opengl", None) or platform_supports_feature(bld.env.PLATFORM, "vulkan", None)):
```
Comment (JCash) [inline engine/engine/src/wscript:138]: For platforms that don't support neither.

File: engine/platform/src/platform_window_null.cpp [hunk 3]
```diff
@@ -118,6 +118,9 @@ namespace dmPlatform
     void PollEvents(HWindow window)
     {}
 
+    void SwapBuffers(HWindow window)
+    {}
+
```
Comment (JCash) [inline engine/platform/src/platform_window_null.cpp:123]: We should add a test to make sure it links (like we have for physics_null library)

File: scripts/build.py [hunk 4]
```diff
@@ -2280,6 +2278,10 @@ def _form_env(self):
         env['PYTHONPATH'] = os.path.pathsep.join(pythonpaths)
         env['PYTHONIOENCODING'] = 'UTF-8'
 
+        if not 'JAVA_HOME' in os.environ:
+            self.find_and_set_java_home()
+        if not 'JAVA_HOME' in os.environ:
+            self.fatal("Failed to find JAVA_HOME environment variable or valid java executable")
```
Comment (JCash) [inline scripts/build.py:2284]: Helped pick up my local javac installation

**Review by Jhonnyg — APPROVED** [review]

**Pull request merged**